### PR TITLE
Improve Webpack Configuration

### PR DESCRIPTION
**Pull request by jtoar**
> Link to [Webpack Configuration](https://deploy-preview-219--redwoodjs.netlify.app/docs/webpack-configuration)

This PR improves the Webpack Configuration by adding more explanation and examples.

Also keeping on eye on https://github.com/redwoodjs/redwood/issues/534#issuecomment-650471654. It's looking like I'll incorporate those changes here or in another PR.

### CODE REVIEW

File: docs/webpackConfiguration.md [hunk 1]
```diff
@@ -1,27 +1,108 @@
...
+<!-- So the settings here apply to both. -->
+
+Here's a table of the plugins we use:
+
+| Plugin                                                                                                                                                                                    | Description                                                                                 |
+| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------ |
+| [HtmlWebpackPlugin](https://webpack.js.org/plugins/html-webpack-plugin/)                                                                                                                  | Simplifies the creation of HTML files to serve your webpack bundles                         |
+| [DirectoryNamedWebpackPlugin](https://github.com/shaketbaby/directory-named-webpack-plugin#readme): Makes it possible to control what file within directory will be treated as entry file |
```
Comment (bjackson) [inline docs/webpackConfiguration.md:34]:
```suggestion
| [DirectoryNamedWebpackPlugin](https://github.com/shaketbaby/directory-named-webpack-plugin#readme) | Makes it possible to control what file within directory will be treated as entry file |
```

Comment (jtoar) [reply docs/webpackConfiguration.md:34]: Thank you!

**Comment by peterp** [comment]
Wooh, this is awesome!

File: docs/webpackConfiguration.md [hunk 2]
```diff
@@ -31,11 +120,40 @@ module.exports = {
 }
 ```
 
-## Example Use
+<!-- TODO -->
+<!-- Check this... -->
+There's a couple reasons for using one format over the other.
+You can only configure a speicifc environment (i.e. development, production) using the first format.
```
Comment (peterp) [inline docs/webpackConfiguration.md:126]:
```suggestion
You can only configure a specific environment (i.e. development, production) using the first format.
```

File: docs/webpackConfiguration.md [hunk 3]
```diff
@@ -31,11 +120,40 @@ module.exports = {
...
+And as the [Changing the Title of the Page](#changing-the-title-of-the-page) example shows, you can also only modify existing rules using the first format.
 
-### Changing the title of the page
+If you're adding extra things, the second format should work just fine.
+
+## Examples
+
+### Changing the Title of the Page
```
Comment (peterp) [inline docs/webpackConfiguration.md:133]: Ideally I think we should ask them to modify `web/index.html` since it's way easier to change the title.

Comment (jtoar) [reply docs/webpackConfiguration.md:127]: I added a note saying that it's much easier to do that (go to web/index.html), but left the section as an example of how to search the plugins array to change a setting.

**Pull request merged**